bentoml/_internal/frameworks/onnxmlir.py

```diff
@@ -0,0 +1,396 @@
+
+    Example:
+
+    .. code-block:: python
+
+       import bentoml
+
+       model = bentoml.onnxmlir.save_model("mlir_mnist", "/path/to/mnist.onnx")
+
+    If you compile the model yourself, you can pass in :code:`trust_compiled_model=True` to save the compiled model directly:
+
+    .. code-block:: python
+
+       import bentoml
+
+       model = bentoml.onnxmlir.save_model("mlir_mnist", "./mnist.so", trust_compiled_model=True)
+
+    To use other platform, you can pass in :code:`docker_platform`:
+
+    .. code-block:: python
+
+       # compile model on s390x platform
+       import bentoml
+
+       model = bentoml.onnxmlir.save_model("mlir_mnist", "./mnist.onnx", docker_platform="s390x")
+    """
+    context = ModelContext(
+        framework_name="onnxmlir",
+        framework_versions={"onnxmlir": _get_onnxmlir_version()},
+    )
+
+    if signatures is None:
+        signatures = {
+            "run": {"batchable": False},
+        }
+        logger.info(
+            f"Using the default model signature for onnxmlir ({signatures}) for model {name}."
+        )
+    if not isinstance(model_path, Path):
+        model_path = Path(model_path)
+
+    if model_path.suffix != ".onnx":
+        if model_path.suffix == ".so":
+            if not trust_compiled_model:
+                raise BentoMLException(
+                    "'trust_compiled_model' must set to True when saving a compiled model."
+                )
+        else:
+            raise BentoMLException("given 'model_path' has to be a ONNX model format.")
+    model_path = os.path.abspath(os.path.expanduser(model_path))
+
+    options = ONNXMLirOptions(
+        use_default_entry_point=use_default_entry_point, use_docker=not ONNX_MLIR
+    )
+
+    with bentoml.models.create(
+        name,
+        module=MODULE_NAME,
+        api_version=API_VERSION,
+        signatures=signatures,
+        labels=labels,
+        custom_objects=custom_objects,
+        options=options,
+        metadata=metadata,
+        context=context,
+    ) as bento_model:
+        if trust_compiled_model and model_path.endswith(".so"):
+            shutil.copyfile(model_path, bento_model.path_of(MODEL_FILENAME))
+            return bento_model
+        if ONNX_MLIR:
+            subprocess.check_output(
+                [
+                    ONNX_MLIR,
+                    *COMPILE_ARG,
+                    model_path,
+                    "-o",
+                    bento_model.path_of(MODEL_FILENAME.split(".")[0]),
+                ]
+            )
+        else:
+            # docker run --rm -it -v $(pwd):/workdir onnxmlirczar/onnx-mlir:latest -O3 --EmitLib /workdir/docs/mnist_example/mnist.onnx -o torch_mnist
+            logger.info(
+                f"Using {RELEASE_MLIR_IMAGE} to compile given model {model_path}. It might take a while if this is the first time you download {RELEASE_MLIR_IMAGE}."
+            )
```

Comment:
In this case, we can only run docker images on x86, ppc64le, and s390x. I can't seem to make the compiled model to run on arm ;(